### PR TITLE
[cpprealm] New Port

### PR DESCRIPTION
**Pull request by leemaguire**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [ ] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/cpprealm/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        ${MATCHED_RELEASE_TYPE_HEADER_CONTENTS}
+        #endif
+}
+    " MODIFIED_HEADER "${DEBUG_TYPE_HEADER_CONTENTS}")
+
+file(WRITE ${CURRENT_PACKAGES_DIR}/include/cpprealm/internal/bridge/bridge_types.hpp "${MODIFIED_HEADER}")
+
+file(REMOVE_RECURSE
+    "${CURRENT_PACKAGES_DIR}/debug/include"
+    "${CURRENT_PACKAGES_DIR}/debug/share"
```
Comment (Cheney-W) [inline ports/cpprealm/portfile.cmake:90]:
```suggestion
    "${CURRENT_PACKAGES_DIR}/debug/share"
    "${CURRENT_PACKAGES_DIR}/debug/doc"
    "${CURRENT_PACKAGES_DIR}/doc"
```

File: ports/cpprealm/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    SOURCE_PATH "${SOURCE_PATH}"
+    DISABLE_PARALLEL_CONFIGURE
+    OPTIONS
+        ${CPPREALM_CMAKE_OPTIONS}
+)
+
+vcpkg_cmake_install()
+
+vcpkg_cmake_config_fixup(PACKAGE_NAME "cpprealm" CONFIG_PATH "cmake")
+vcpkg_cmake_config_fixup(PACKAGE_NAME "Realm" CONFIG_PATH "share/cmake/Realm")
```
Comment (Cheney-W) [inline ports/cpprealm/portfile.cmake:64]:
```suggestion
vcpkg_cmake_config_fixup(PACKAGE_NAME "realm" CONFIG_PATH "share/cmake/Realm")
```

File: ports/cpprealm/usage [hunk 3]
```diff
@@ -0,0 +1,4 @@
+cpprealm provides CMake targets:
+
+    find_package(cpprealm CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE cpprealm)
```
Comment (Cheney-W) [inline ports/cpprealm/usage:4]: Don't you need to provide the usage of realm here?

**Comment by Cheney-W** [comment]
Please follow the robot prompts to sign Contributor License Agreement(CLA).

**Comment by leemaguire** [comment]
@microsoft-github-policy-service agree [company="MongoDB"]

**Comment by leemaguire** [comment]
@microsoft-github-policy-service agree company="MongoDB"

File: ports/cpprealm/usage (hunk 3 above)
Comment (leemaguire) [reply ports/cpprealm/usage:4]: I followed the usage template for this, everything required to use `cpprealm` in CMake is listed above.

**Review by Cheney-W — APPROVED** [review]

**Comment by Cheney-W** [comment]
Usage test passed with `x64-windows` triplet.

**Review by JavierMatosD — COMMENTED** [review]
All dependencies have to come from the registry and no port should [vender their own dependencies](https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#do-not-use-vendored-dependencies). If they are not in the registry, they should be added in [separate PRs per port]( https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#make-separate-pull-requests-per-port).

So far I see this port providing the following libraries under various conditions:

- `realm-cpp`
- `realm-core`
- `Catch2`
- `sha1`
- `sha-2`

We should also be using upstreams name unless it is not in repology.org. Why rename the port from `realm-cpp` to `cpprealm`? 

I also see you are using string replacement to patch over some of the headers? Can you explain your rationale?

**Comment by leemaguire** [comment]
> All dependencies have to come from the registry and no port should [vender their own dependencies](https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#do-not-use-vendored-dependencies). If they are not in the registry, they should be added in [separate PRs per port](https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#make-separate-pull-requests-per-port).
> 
> So far I see this port providing the following libraries under various conditions:
> 
> * `realm-cpp`
> * `realm-core`
> * `Catch2`
> * `sha1`
> * `sha-2`
> 
> We should also be using upstreams name unless it is not in repology.org. Why rename the port from `realm-cpp` to `cpprealm`?
> 
> I also see you are using string replacement to patch over some of the headers? Can you explain your rationale?

Hi @JavierMatosD

Thank you for your feedback. I understand your concerns. To address the former, I will close this PR and first submit a new one specifically for the realm-core dependency, followed by another for realm-cpp.

Our target name is `cpprealm`, while the repo is named realm-cpp. Changing this would be a breaking change for us. We are not currently in repology.org.

The string replacements in the headers are necessary because we use `std::aligned_storage` to encapsulate realm-core types within our bridge classes. Since we cannot publicly expose these types, we must compute and inject the correct size and alignment values during installation. This requirement leads us to modify the bridge_types.hpp header as part of our build process.

**Pull request closed without merging**